packages/relay/src/Environment.js

```diff
@@ -1,13 +1,54 @@
 // @flow
 
 import { GetDeviceLocale } from '@kiwicom/mobile-localization';
-import { Environment, Network, RecordSource, Store } from 'relay-runtime';
+import {
+  Environment,
+  Network,
+  RecordSource,
+  Store,
+  Observable,
+  createOperationSelector,
+} from 'relay-runtime';
+import { AsyncStorage } from 'react-native';
 
 type FetcherResponse = {|
   +data: Object,
   +errors?: $ReadOnlyArray<Object>,
 |};
 
+const ASYNC_STORE_KEY = '@OfflineStore:key';
+const GRAPHQL_URL = 'https://graphql.kiwi.com/';
+
+const store = new Store(new RecordSource());
+
+function fetchFromTheNetwork(networkHeaders, operation, variables, observer) {
+  fetch(GRAPHQL_URL, {
+    method: 'POST',
+    headers: networkHeaders,
+    body: JSON.stringify({
+      query: operation.text, // TODO: fetch persisted queries instead (based on operation.id)
+      variables,
+    }),
+  })
+    .then(fetchResponse => fetchResponse.json())
+    .then(jsonResponse => {
+      if (jsonResponse.errors) {
+        jsonResponse.errors.forEach(error => console.warn(error));
+      }
+      observer.next(jsonResponse);
```

Comment:
If I understand correctly, this observer will send `jsonResponse` to the store and it will be stored there.

Reply:
No, this is like `return` from the fetch (it's more like a `yield` here). This way I can return multiple values (first from the cache and later from the network). Components will be re-rendered only if needed.